In [1]:
## Sympy stuff

from sympy import *
x = symbols('x')
y = symbols('y')

<div class="page_break"> </div>

# Areas between curves
**([Syllabus Unit 3](https://canvas.ubc.ca/courses/92840/syllabus) | [CLP2 Chapter 1.5](https://personal.math.ubc.ca/~CLP/CLP2/))**



## Introduction

### About Me

[<img src="images/firas_profile.png" width=100%>](https://firas.moosvi.com)


### Announcements

- HW03 is now open

<div class="page_break"> </div>

### Area under a curve

**Recall**: If $f$ is integrable and $f \geq 0$ on the interval $[a,b]$,

\begin{align}
\int_{a}^{b}f(x) dx= A,
\end{align}

this Area $A$, lies above $y=0$ and below $y=f(x)$, in the vertical strip $a\le x\le b.$

<img src="images/curve_introA.png" width="70%">

### General case: area between two curves

**Today**: We can extend this to a more general situation by replacing the $x$-axis ($y=0$) with a general lower boundary $y=g(x)$. 

<img src="images/curve_introB.png" width="70%">

*Note: Setting $g(x)=0$ will recapture the original definition.*

### Leibniz-style shorthand: 

$A = \int\!dA$, with $dA = (f(x)-g(x))\,dx$ over $a\leq x\leq b$.

<div class="page_break"> </div>

## An intuitive description

In [2]:
%%html
<iframe scrolling="no" title="Area Between Two Curves" src="https://www.geogebra.org/material/iframe/id/y6p3qybd/width/1680/height/885/border/888888/sfsb/true/smb/false/stb/false/stbh/false/ai/false/asb/false/sri/false/rc/false/ld/false/sdz/true/ctl/false" width="100%" height="800px" style="border:0px;"> </iframe>

<div class="page_break"> </div>

## Motivation

- Often when we're in school, and going through the rigours of daily life, we lose sight of the bigger picture, the important things in life.
- Things like social justice, fighting for equity, righting the wrongs of the world, and making the world a better place.
- That is why, I take every opportunity to make connections between what we're doing in class, and the real world.
- Before we get into the mechanics of computing areas between curves, let's look at this practical example.

<img src="images/gini.png">

[Source: Our World in Data](https://ourworldindata.org/income-inequality#the-gini-coefficient)

### Additional Reading

- [Applications of Area Between Curves, Section 6.5](https://math.hws.edu/~mitchell/Math131F15/tufte-latex/BetwCurves2.pdf)

<div class="page_break"> </div>

## Computing Area under two curves

### Example I.1

```{admonition} Example I.1
:class: seealso
Compute the finite area between the curves $f(x)= x+2$ and $g(x)=x-2$ from $x=0$ to $x=3$.

<img src="images/curve0.png" class="center" width=100%>
```

```{admonition} Strategy
:class: tip
1. Given $f(x)$ and $g(x)$, integrate $f(x) - g(x)$ from $x=a$ to $x=b$ to determine the area between the curves, $A$.

    $$
    A = \int_a^b f(x) - g(x) dx
    $$
```

In [7]:
f = x+2
g = x-2

a = 0 # Lower x-limit in the domain of interest
b = 3 # Upper x-limit in the domain of interest

In [8]:
integrate(f)

x**2/2 + 2*x

In [10]:
integrate(g)

x**2/2 - 2*x

In [11]:
integrate(f) - integrate(g) # integrate f(x) - g(x)

4*x

In [13]:
integrate(f-g,x) # Note, this gives the same answer as above!

4*x

In [14]:
integrate(f,(x,0,3)) # integrate f(x) with x ranging from 0 to 3

21/2

In [16]:
integrate(g,(x,0,3)) # integrate g(x) with x ranging from 0 to 3

-3/2

In [17]:
# Final answer

A_1 = integrate(f,(x,0,3)) - integrate(g,(x,0,3))

The Area between $f(x) = x+2$ and $g(x) = x-2$ , between x=0 and x=3 is:

In [16]:
A_1

12

<div class="page_break"> </div>

## Consideration 1: Does it matter which function is greater?

```{admonition} Example I.1
:class: seealso
Compute the finite area between the curves $f(x)= x+2$ and $g(x)=x-2$ from $x=0$ to $x=3$.

<img src="images/curve0.png" class="center" width=100%>
```

In the previous example, what if we had gotten the order wrong, and done

$$
A = \int_a^b g(x) - f(x) dx
$$

instead?

In [18]:
integrate(g,(x,0,3)) - integrate(f,(x,0,3))

-12

Interesting! It seems that if we get the order wrong, our answer will be negative.

Intuitively, this makes sense. By convention, we will require that areas in the first quadrant (top-right of an $x-y$ axes) will be **positive**.

Let's update our strategy given what we just discovered:

```{admonition} Strategy
:class: tip
1. **Given $f(x)$ and $g(x)$, determine which function is greater.**

2. **Then, assuming $f(x)$ is greater**, integrate $f(x) - g(x)$ from $x=a$ to $x=b$ to determine the area between the curves, $A$.

    $$
    A = \int_a^b f(x) - g(x) dx
    $$
```

## Consideration 2: Solving problems where the functions intersect.

What if the curves $y=f(x)$ and $y=g(x)$ cross somewhere in the given interval?

*Hint*: Remember that “area” is always a positive quantity!

### Example I.2A

```{admonition} Example I.2A
:class: seealso
Compute the finite area between where the curves $f(x)= x+2$ and $g(x)=x^2-4$ intersect.

<img src="images/curve2v2.png" class="center" width=100%>
```

In [18]:
f = x + 2
g = x**2 - 4

# a and b are currently unknown! We must set f = g and solve to figure out the roots.

In [19]:
a,b = solve(f-g)
display(a,b)

-2

3

In [20]:
integrate(f)

x**2/2 + 2*x

In [21]:
integrate(g)

x**3/3 - 4*x

In [22]:
integrate(f,(x,a,b))

25/2

In [23]:
integrate(g,(x,a,b))

-25/3

In [24]:
# Final answer

A_2 = integrate(f,(x,a,b)) - integrate(g,(x,a,b))

The area between $x + 2$ and $x^2 - 4$ between $x=-2$ and $x=4$:

In [25]:
A_2

125/6

Let's update our strategy given what we just discovered:

```{admonition} Strategy
:class: tip

1. **Roughly sketch the curves $f(x)$ and $g(x)$.**

2. **Determine the domain of interest; usually by setting $f(x) = g(x)$ and solving for the roots. Though, other operations may be needed as well depending on the question.**

3. Given $f(x)$ and $g(x)$, determine which function is greater **within the domain of interest**.

4. Then, assuming $f(x)$ is greater, integrate $f(x) - g(x)$ from $x=a$ to $x=b$ to determine the area between the curves, $A$.

    $$
    A = \int_a^b f(x) - g(x) dx
    $$
```

<div class="page_break"> </div>

## Consideration 3: What if the functions cross, and the "greater function" changes?

### Example I.2B

```{admonition} Example I.1
:class: seealso
Compute the finite area between the curves $f(x)= x+2$ and $g(x)=x^2-4$ between $x=-2$ and $x=4$.

<img src="images/curve2bv2.png" class="center" width=100%>
```

Notice that we have already computed the area between $f(x)$ and $g(x)$ between $x=-2$ and $x=3$, we called that $A_2$.

To find the area between $f(x)$ and $g(x)$ between $x=-2$ and $x=4$, we first notice that we need to split our integral into two pieces:

1. Integral between $x=-2$ and $x=3$ (where $f(x)$ is greater):

$$
A_{3a} = \int_{-2}^{+3} [f(x) - g(x)]dx = A_2
$$

**and (added to)**

2. Integral between $x=3$ and $x=4$ (where $g(x)$ is greater):

$$
A_{3b} = \int_{3}^{4} [g(x) - f(x)]dx
$$

The combined integral is then:

$$
A_3 = A_{3a} + A_{3b}
$$

In [26]:
# Final answer

a = 3
b = 4

A_3 = A_2 + integrate(f,(x,a,b)) - integrate(g,(x,a,b))

The area between $x + 2$ and $x^2 - 4$ between $x=-2$ and $x=4$:

In [27]:
A_3

18

Let's update our strategy given what we just discovered:

```{admonition} Strategy
:class: tip

1. Roughly sketch the curves $f(x)$ and $g(x)$.

2. Determine the domain of interest; usually by setting $f(x) = g(x)$ and solving for the roots. Though, other operations may be needed as well depending on the question.

3. Given $f(x)$ and $g(x)$, determine which function is greater within the domain(s) of interest. **If the curves cross, split the integral so each segment is considered separately.**

4. **Region 1: Assuming $f(x)$ is greater in region 1, integrate $f(x) - g(x)$ from $x=a$ to $x=b$.**

    $$
    A_1 = \int_a^b f(x) - g(x) dx
    $$

5. **Region 2: If there are one or more function crosses, there will be multiple segments to integrate. Now, $g(x)$ will be greater in region 2, so determine $A_2$:**

    $$
    A_2 = \int_c^d g(x) - f(x) dx
    $$

6. **Compute integrals of each segment and after after each cross. Add all the area segments together:**

    $$
    A_{total} = A_1 + A_2 +... A_N
    $$
```

<div class="page_break"> </div>

## Consideration 4: Should we integrate over $x$, or over $y$?

When the upper or lower boundary curve has a piecewise definition, creative geometry-inspired thinking can offer an efficient new approach. Here is an example:

### Example I.3

```{admonition} Example I.1
:class: seealso
Find the finite area between the curves $y^2= 4x$ and $4x-3y=4$.

<img src="images/curve1v2.png" class="center" width=100%>
```

First, we need to re-write $y^2= 4x$ as $f(x)$, and and $4x-3y=4$ as $g(x)$.

With a bit of algebra,

$$
f(x) = \sqrt{4x}
$$
$$
g(x) = \frac{4(x-1)}{3}
$$

So let's figure out $a$ and $b$ using the same strategy we used earlier, set $f(x) = g(x)$ and solve for the roots:

In [28]:
f = sqrt(4*x)     # Let's call the top half of the parabola +f and the bottom half -f
g = (4/3) * (x-1)

a = solve((-f)-g)[0]
b = solve((+f)-g)[0]

In [29]:
print(f"a = {a:.2f}\nb = {b:.0f}")

a = 0.25
b = 4


Now, we have two options.

- We can integrate over $x$ with vertical rectangles of width $dx$.

Or, 

- We can integrate over $y$ with horizontal rectangles of height $dy$.

<div class="page_break"> </div>

### Option 1: Integrate over x (the long way)

We can continue doing what we've done so far and integrate over $x$ with vertical rectangles.

<img src="images/curve1v2_vertical.png">



First, we need to setup the limits of our integral. Note that there is something a bit tricky here when we go to determine the "greater" and "lower" curves.

- Between $x=0$ and the point $x=0.25$ ($a$), where $f(x)$ and $g(x)$ first intersect, the "greater" curve is the upper half of the parabola $f(x) = +\sqrt{4x}$, and the "lower" curve is the lower half of the parabola $f(x) = -\sqrt{4x}$.

- Between $x=0.25$ ($a$) and the second point of intersection $x=4$ ($b$), the "greater" curve is the upper half of the parabola $f(x)$, and the "lower" curve is $g(x)$.

So we need to split up our integral in two parts:

\begin{align}
A_{4x} = A_{4A} + A_{4B}\\
A_{4x} = \int_0^{a} (g(x)-f(x))dx + \int_a^{b} (f(x) - g(x))dx
\end{align}

In [30]:
A_4A = integrate(f--f,(x,0,a))

In [31]:
A_4A

0.333333333333333

In [32]:
A_4B = integrate(f-g,(x,a,b))

In [33]:
A_4B

4.87500000000000

In [34]:
A4x = A_4A + A_4B

The Area between $f(x) = \pm \sqrt{4x}$ and $g(x) = \frac{4(x-1)}{3}$ between $x =0.25$ and $x = 4$ is:

In [35]:
A4x

5.20833333333333

or, equivalently,

$$
\frac{125}{24}.
$$

Phew! This was a bit painful, is there a faster way?

<div class="page_break"> </div>

### Option 2: Integrate over $y$ (the quick way!)

Looking at the sketch, it looks like we can also integrate over $y$, with horizontal rectangles.

<img src="images/curve1v2_horiz.png">

The advantage of integrating over $y$, is that our integral is quite simple, because the parabola $f(x)$ is greater than $g(x)$ within our domain of interest ($x=0.25$ and $x=4$).

With a bit of algebra, we can rearrange $f(x)$ and $g(x)$ to write $f(y)$ and $g(y)$:

$$
f(y) = \frac{y^2}{4}
$$

and,

$$
g(y) = \frac{3y}{4} + 1.
$$

Since we are interested in the area between $f(x)$ and $g(x)$, the curves are bounded by the $(x,y)$ points $(a,c)$ on the left, and $(b,d)$ on the right.
We already know $a$ and $b$, what remains is to determine $c$ and $d$, the $y$-coordinates at the intersection points.

$$
A_{4y} = \int_c^{d} (g(y)-f(y))dy
$$

Because our bounds are at the points where $f(y)$ and $g(y)$ intersect, we're in luck and we can use either function to determine the corresponding $y$-values $c$ and $d$.

Let's use $g(x)$:

$$
g(x) = \frac{4(x-1)}{3}.
$$

Substituting in $a=0.25$ on the left side,
$$
c = g(0.25) = \frac{4(0.25-1)}{3} + 1
$$

we get $c = -1$.

On the right side,

$$
d = g(b)
$$

Substituting in $b=4$,
$$
d = \frac{4(4-1)}{3}
$$

we get $d = 4$.

Now that we have the limits of the $y$-integration, we can complete the integral:

$$
A_{4y} = \int_{c=-1}^{d=4} (f(y) - g(y)) dy
$$

In [36]:
c = -1
d = 4

f_y = y**2/4
g_y = 3*y/4 + 1

A_4y = integrate(g_y-f_y,(y,c,d))

The Area between $f(y) = \frac{y^2}{4}$ and $g(y) = \frac{3y}{4} + 1$ between $y = -1$ and $y = 4$ is:

In [37]:
A_4y

125/24

Great! As expceted, it's the same answer as we got if we integrated over $x$ instead of $y$.
A little bit less painful, but required a bit of foresight and a good sketch of the functions!

Let's update our Strategy one last time given what we just discovered...

```{admonition} Strategy
:class: tip

1. Roughly sketch the curves $f(x)$ and $g(x)$.

2. Determine the domain of interest; usually by setting $f(x) = g(x)$ and solving for the roots. Though, other operations may be needed as well depending on the question.

3. **Decide if it is easier to integrate over $x$ (vertical rectangles) or $y$ (horizontal rectangles). If integrating over $y$, determine $f(y)$ and $g(y)$, and the *ranges* of interest (instead of the domains, if integrating over $x$).**

4. Given $f(x)$ and $g(x)$ (**or $f(y)$ and $g(y)$**), determine which function is greater within the domain(s) (**or range(s)**) of interest. If the curves cross, split the integral so each segment is considered separately.

    **Below this, if integrating over $y$, replace $f(x)$ with $f(y)$, $dx$ with $dy$, etc...**

5. Region 1: Assuming $f(x)$ is greater in region 1, integrate $f(x) - g(x)$ from $x=a$ to $x=b$. 

    $$
    A_1 = \int_a^b f(x) - g(x) dx
    $$

6. Region 2: If there are one or more function crosses, there will be multiple segments to integrate. Now, $g(x)$ will be greater in region 2, so determine $A_2$:

    $$
    A_2 = \int_c^d g(x) - f(x) dx
    $$

7. Compute integrals of each segment and after after each cross. Add all the area segments together:

    $$
    A_{total} = A_1 + A_2 +... A_N
    $$
```

<div class="page_break"> </div>

<div class="page_break"> </div>

## Summary

1. Extending definition of "Area under curve" between $f(x)$ and $y=0$, to determine area between $f(x)$ and $g(x)$.

2. Discussing income inequality and a practical application of computing area between curves.

3. Mechanics of computing areas between curves, including determining intersection points and the domain or range of integration.

4. Consideration 1: Does it matter which function is greater?

5. Consideration 2: Solving problems where the functions intersect.

6. Consideration 3: What if the functions cross, and the "greater function" changes?

7. Consideration 4: Should we integrate over $x$, or over $y$?

8. Development of working strategy...

Here is the final strategy that we created together to solve area between curve problems.

*Note: You will need to add or revise this strategy as you learn more skills and tricks, and become more familiar with the mechanics of doing integrals!*

```{admonition} Strategy
:class: tip

1. Roughly sketch the curves $f(x)$ and $g(x)$.

2. Determine the domain of interest; usually by setting $f(x) = g(x)$ and solving for the roots. Though, other operations may be needed as well depending on the question.

3. **Decide if it is easier to integrate over $x$ (vertical rectangles) or $y$ (horizontal rectangles). If integrating over $y$, determine $f(y)$ and $g(y)$, and the *ranges* of interest (instead of the domains, if integrating over $x$).**

4. Given $f(x)$ and $g(x)$ (**or $f(y)$ and $g(y)$**), determine which function is greater within the domain(s) (**or range(s)**) of interest. If the curves cross, split the integral so each segment is considered separately.

    **Below this, if integrating over $y$, replace $f(x)$ with $f(y)$, $dx$ with $dy$, etc...**

5. Region 1: Assuming $f(x)$ is greater in region 1, integrate $f(x) - g(x)$ from $x=a$ to $x=b$. 

    $$
    A_1 = \int_a^b f(x) - g(x) dx
    $$

6. Region 2: If there are one or more function crosses, there will be multiple segments to integrate. Now, $g(x)$ will be greater in region 2, so determine $A_2$:

    $$
    A_2 = \int_c^d g(x) - f(x) dx
    $$

7. Compute integrals of each segment and after after each cross. Add all the area segments together:

    $$
    A_{total} = A_1 + A_2 +... A_N
    $$
```

## You Try!

Your post-class activity is to solve this problem. 

### Example I.4

```{admonition} Example I.2
:class: seealso
Find the area of the region bounded by the curves $y=\sin x$, $y=\cos x$, $x=0$, and $x=\pi /2$.

<img src="images/curve3v3.png" class="center" width=100%>
```
*Hint, the answer is not 0!*

Thanks for your time and attention today!

- Dr. Moosvi